packages/flutter/lib/src/foundation/constants.dart

```diff
@@ -38,3 +38,12 @@ const bool kDebugMode = !kReleaseMode && !kProfileMode;
 /// precision loss in calculations. Differences below this threshold are safe to
 /// disregard.
 const double precisionErrorTolerance = 1e-10;
+
+/// A constant that is true if the application was compiled to run on the web.
+///
+/// This implementation takes advantage of the fact that JavaScript does not
+/// support integers. In this environment, Dart's doubles and ints are
+/// backed by the same kind of object. Thus a double `0.0` is identical
+/// to an integer `0`. This is not true for Dart code running in AOT or on the
+/// VM.
+const bool kIsWeb = identical(0, 0.0);
```

Comment:
Should this be `kIsJs`?

Reply:
Technically that is correct, but I worry It might not be as discoverable though.

Reply:
Actually, now that I think about it, we might need both. When we add `kIsJs`, we may want to revise the implementation of `kIsWeb`. One may not imply the other, in any direction.